fiona/_geometry.pyx

```diff
@@ -143,20 +146,49 @@ cdef class GeomBuilder:
 
     cdef build(self, void *geom):
         # The only method anyone needs to call
-        if geom == NULL:
-            raise ValueError("Null geom")
-
-        cdef unsigned int etype = OGR_G_GetGeometryType(geom)
+        cdef void *cogr_geometry = NULL
+        cdef void *cloned_geometry = NULL
 
-        self.code = base_geometry_type_code(etype)
+        if geom == NULL:
+            return None
+
+        code = base_geometry_type_code(OGR_G_GetGeometryType(geom))
+
+        # We convert special geometries (curves, TIN, Triangle, ...)
+        # to GeoJSON compatible geometries (LineStrings, Polygons, MultiPolygon, ...)
+        if 8 <= code <= 14:  # Curves.
+            # OGR_G_GetLinearGeometry: The ownership of the returned geometry belongs to the caller.
+            cogr_geometry = OGR_G_GetLinearGeometry(geom, 0.0, NULL)
+            code = base_geometry_type_code(OGR_G_GetGeometryType(cogr_geometry))
+            ndims = OGR_G_GetCoordinateDimension(cogr_geometry)
+            geom = cogr_geometry
+        elif 15 <= code <= 17:
+            cloned_geometry = OGR_G_Clone(geom)
```

Comment:
Without cloning the geometry, (respectively initially stealing the geometry from the feature the code segfaults). I'm not entirely sure yet why.

Reply:
Thanks. This explains it!